### PR TITLE
Support dot in IDs

### PR DESCRIPTION
**Pull request by aferr**
Because antlr does not allow lexer rules that overlap, the PATH lexer
rule is replaced with ID which is now used in both positions. This means
that quotes can appear in the names of file paths (which is not desired), but since this code
is likely to be thrown out, this is probably OK.

### CODE REVIEW

**Comment by bgogul** [comment]
> that quotes can appear in the names of file paths (which is not desired)

Why is undesirable? Shouldn't it be OK to say `import "some.file.auth"`?

File: rust/tools/authorization-logic/src/parsing/AuthLogic.g4 [hunk 1]
```diff
@@ -60,22 +60,25 @@ assertion
     | fact ':-' flatFact (',' flatFact )* '.' #hornClauseAssertion
     ;
 
+// The IDs following "Export" are path names where JSON files containing
```
Comment (bgogul) [inline rust/tools/authorization-logic/src/parsing/AuthLogic.g4:63]: Can you add a test?

**Review by bgogul — APPROVED** [review]

**Comment by aferr** [comment]
> > that quotes can appear in the names of file paths (which is not desired)
> 
> Why is undesirable? Shouldn't it be OK to say `import "some.file.auth"`?

You're right I guess that's fine.

File: rust/tools/authorization-logic/src/parsing/AuthLogic.g4 (hunk 1 above)
Comment (aferr) [reply rust/tools/authorization-logic/src/parsing/AuthLogic.g4:63]: There is already a test that checks that policy statements can be properly imported and exported and some tests about the related cryptographic signature checking.

I am guessing that I am over-indexing on the line you are commenting on, and that you want a test of the new functionality from this commit.

Comment (bgogul) [reply rust/tools/authorization-logic/src/parsing/AuthLogic.g4:63]: I meant adding a test where access path has a dot (`.`) in it. I don't think any of the test_inputs meet the criteria?

**Review by markww — APPROVED** [review]

**Pull request merged**